compiler/fme-detect/src/EqualizePatternFinder.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "EqualizePatternFinder.h"
+
+#include <luci/IR/CircleNode.h>
+#include <luci/IR/CircleNodeVisitor.h>
+
+#include <vector>
+
+using namespace fme_detect;
+
+namespace
+{
+
+struct Fusability
+{
+  bool pre_scale = false;
+  bool post_scale = false;
+  bool pre_shift = false;
+  bool post_shift = false;
+};
+
+struct GetFusability final : public luci::CircleNodeMutableVisitor<Fusability>
```

Comment:
How about spliting this part..? To avoid frequent merge conflict in near future?